taxi/views.py

```diff
@@ -34,16 +41,34 @@ class ManufacturerListView(LoginRequiredMixin, generic.ListView):
     model = Manufacturer
     context_object_name = "manufacturer_list"
     template_name = "taxi/manufacturer_list.html"
-    paginate_by = 5
+    paginate_by = 2
+    queryset = Manufacturer.objects.all()
 
+    def get_context_data(self, *, object_list=None, **kwargs):
+        context = super(ManufacturerListView, self).get_context_data(**kwargs)
+        context["search_form"] = ManufacturerSearchForm()
+        return context
 
-class ManufacturerCreateView(LoginRequiredMixin, generic.CreateView):
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        form = ManufacturerSearchForm(self.request.GET)
+
+        if form.is_valid():
+            name = form.cleaned_data.get("name")
+            if name:
+                queryset = queryset.filter(name__icontains=name)
+        return queryset
+
+
+class ManufacturerCreateView(LoginRequiredMixin,
+                             generic.CreateView):
```

Comment:
The `ManufacturerCreateView` class is missing a form class assignment. It should specify a form class if any specific form is intended to be used for creating a manufacturer.